code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleExtension.java

```diff
@@ -0,0 +1,239 @@
+		lifecycleDataStore = namedCollection;
+		this.deviceInfoService = deviceInfoService;
+		this.lifecycleV1 = lifecycleV1Extension;
+		lifecycleV2 = lifecycleV2Extension;
+	}
+
+	@Override
+	protected String getName() {
+		return LifecycleConstants.EventDataKeys.Lifecycle.MODULE_NAME;
+	}
+
+	@Override
+	protected String getVersion() {
+		return Lifecycle.extensionVersion();
+	}
+
+	@Override
+	protected String getFriendlyName() {
+		return LifecycleConstants.FRIENDLY_NAME;
+	}
+
+	@Override
+	protected void onRegistered() {
+		getApi().registerEventListener(EventType.GENERIC_LIFECYCLE, EventSource.REQUEST_CONTENT, this::handleLifecycleRequestEvent);
+		getApi().registerEventListener(EventType.WILDCARD, EventSource.WILDCARD, this::updateLastKnownTimestamp);
+		lifecycleV1.processLifecycleExtensionRegistration();
+	}
+
+	@Override
+	public boolean readyForEvent(Event event) {
+		if (event.getType().equalsIgnoreCase(EventType.GENERIC_LIFECYCLE) && event.getSource().equalsIgnoreCase(EventSource.REQUEST_CONTENT)) {
+			SharedStateResult configurationSharedState = getApi().getSharedState(LifecycleConstants.EventDataKeys.Configuration.MODULE_NAME, event, false, SharedStateResolution.ANY);
+			if (configurationSharedState != null) {
+				return configurationSharedState.status == SharedStateStatus.SET;
+			} else {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Processes an event of type generic lifecycle and source request content
+	 * @param event lifecycle request content {@code Event}
+	 */
+	void handleLifecycleRequestEvent(final Event event) {
+		SharedStateResult configurationSharedState = getApi().getSharedState(LifecycleConstants.EventDataKeys.Configuration.MODULE_NAME,
+				event,
+				false,
+				SharedStateResolution.ANY);
+
+		if (configurationSharedState == null ||  configurationSharedState.status == SharedStateStatus.PENDING) {
+			Log.trace(LifecycleConstants.LOG_TAG, "%s - Configuration is pending, waiting...", SELF_LOG_TAG);
+			return;
+		}
+
+		Map<String, Object> eventData = event.getEventData();
+
+		if (eventData == null) {
+			Log.trace(LifecycleConstants.LOG_TAG, "%s - Failed to process lifecycle event '%s for %s'", SELF_LOG_TAG,
+					UNEXPECTED_NULL_VALUE,
+					event.getName());
+			return;
+		}
+
+		String lifecycleAction = (String) eventData.get(LifecycleConstants.EventDataKeys.Lifecycle.LIFECYCLE_ACTION_KEY);
+
+		if (LifecycleConstants.EventDataKeys.Lifecycle.LIFECYCLE_START.equals(lifecycleAction)) {
+			Log.debug(LifecycleConstants.LOG_TAG, "%s - Starting lifecycle", SELF_LOG_TAG);
+			startApplicationLifecycle(event, configurationSharedState.value);
+		} else if (LifecycleConstants.EventDataKeys.Lifecycle.LIFECYCLE_PAUSE.equals(lifecycleAction)) {
+			Log.debug(LifecycleConstants.LOG_TAG, "%s - Pausing lifecycle", SELF_LOG_TAG);
+			pauseApplicationLifecycle(event);
+		} else {
+			Log.warning(LifecycleConstants.LOG_TAG, "%s - Failed to read lifecycle data from persistence", SELF_LOG_TAG);
```

Comment:
Incorrect log message. Can you change it "Missing action in lifecycle request event"